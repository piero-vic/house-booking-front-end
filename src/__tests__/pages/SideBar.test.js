import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import SideBar from '../../pages/Sidebar';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <SideBar />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
