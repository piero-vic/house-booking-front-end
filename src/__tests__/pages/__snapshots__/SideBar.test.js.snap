// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="text-slate-600 sm:w-1/5 z-10 relative max-h-min"
>
  <button
    className="sm:hidden m-3 absolute"
    onClick={[Function]}
    type="button"
  >
    <svg
      className="w-6 h-6 text-gray-500 z-20"
      fill="none"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="4"
      viewBox="0 0 24 24"
    >
      <path
        d="M4 5h16M4 12h16M4"
      />
    </svg>
  </button>
  <nav
    className="hidden sm:flex flex-col w-full sm:w-1/5 h-screen bg-white sm:border-r-2 pt-7 sm:pt-0 sm:fixed"
  >
    <a
      aria-current="page"
      className="m-4 hidden sm:flex active"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="logo"
        src="logo.png"
        width={100}
      />
    </a>
    <ul
      className="sm:ml-2 pt-2 pb-3 mt-6 text-center"
    >
      <li
        className="flex flex-col"
      >
        <a
          aria-current="page"
          className="bg-lime-500 text-white text-sm sm:text-left font-semibold px-3 py-2 hover:bg-lime-500 w-full hover:text-white"
          href="/"
          onClick={[Function]}
        >
          HOUSES
        </a>
        <a
          className="text-sm sm:text-left font-semibold px-3 py-2 hover:bg-lime-500 w-full hover:text-white"
          href="/reservations"
          onClick={[Function]}
        >
          RESERVATIONS
        </a>
        <a
          className="text-sm sm:text-left font-semibold px-3 py-2 hover:bg-lime-500 w-full hover:text-white"
          href="/add_house"
          onClick={[Function]}
        >
          ADD HOUSES
        </a>
        <a
          className="text-sm sm:text-left font-semibold px-3 py-2 hover:bg-lime-500 w-full hover:text-white"
          href="/delete_house"
          onClick={[Function]}
        >
          DELETE HOUSES
        </a>
        <a
          aria-current="page"
          className="text-sm sm:text-left font-semibold px-3 py-2 hover:bg-lime-500 hover:text-white w-full active"
          href="/"
          onClick={[Function]}
        >
          LOGOUT
        </a>
      </li>
    </ul>
    <div
      className="mt-auto text-center"
    >
      <ul
        className="flex justify-center"
      >
        <li>
          <a
            href="https://www.facebook.com"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              baseProfile="tiny"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              version="1.2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13 10h3v3h-3v7h-3v-7h-3v-3h3v-1.255c0-1.189.374-2.691 1.118-3.512.744-.823 1.673-1.233 2.786-1.233h2.096v3h-2.1c-.498 0-.9.402-.9.899v2.101z"
              />
            </svg>
          </a>
        </li>
        <li>
          <a
            href="https://twitter.com/"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              baseProfile="tiny"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              version="1.2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18.89 7.012c.808-.496 1.343-1.173 1.605-2.034-.786.417-1.569.703-2.351.861-.703-.756-1.593-1.14-2.66-1.14-1.043 0-1.924.366-2.643 1.078-.715.717-1.076 1.588-1.076 2.605 0 .309.039.585.117.819-3.076-.105-5.622-1.381-7.628-3.837-.34.601-.51 1.213-.51 1.846 0 1.301.549 2.332 1.645 3.089-.625-.053-1.176-.211-1.645-.47 0 .929.273 1.705.82 2.388.549.676 1.254 1.107 2.115 1.291-.312.08-.641.118-.979.118-.312 0-.533-.026-.664-.083.23.757.664 1.371 1.291 1.841.625.472 1.344.721 2.152.743-1.332 1.045-2.855 1.562-4.578 1.562-.422 0-.721-.006-.902-.038 1.697 1.102 3.586 1.649 5.676 1.649 2.139 0 4.029-.542 5.674-1.626 1.645-1.078 2.859-2.408 3.639-3.974.784-1.564 1.172-3.192 1.172-4.892v-.468c.758-.57 1.371-1.212 1.84-1.921-.68.293-1.383.492-2.11.593z"
              />
            </svg>
          </a>
        </li>
        <li>
          <a
            href="https://myaccount.google.com/"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              version="1"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12.9 13.5l-.7-.5c-.2-.2-.5-.4-.5-.8s.3-.7.6-1c.8-.6 1.7-1.3 1.7-2.8 0-1.5-.9-2.3-1.4-2.7h1.2l1.2-.7h-4.1c-1 0-2.4.2-3.5 1.1-.8.7-1.2 1.7-1.2 2.6 0 1.5 1.2 3.1 3.3 3.1h.6c-.1.2-.2.4-.2.7 0 .6.3 1 .6 1.3-.9.1-2.5.2-3.8.9-1.2.7-1.5 1.7-1.5 2.4 0 1.5 1.4 2.8 4.2 2.8 3.4 0 5.2-1.9 5.2-3.7 0-1.3-.8-1.9-1.7-2.7zm-2.5-2.2c-1.7 0-2.5-2.2-2.5-3.5 0-.5.1-1 .4-1.5.3-.4.9-.7 1.4-.7 1.6 0 2.5 2.2 2.5 3.6 0 .4 0 1-.5 1.4-.3.4-.9.7-1.3.7zm0 7.9c-2.1 0-3.5-1-3.5-2.4s1.3-1.9 1.7-2c.8-.3 1.9-.3 2.1-.3h.5c1.5 1.1 2.1 1.6 2.1 2.6 0 1.2-1 2.1-2.9 2.1zM17 12h-2v-1h2v-1.9l1-.1v2h2v1h-2v2h-1z"
              />
            </svg>
          </a>
        </li>
        <li>
          <a
            href="https://vimeo.com/"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              baseProfile="tiny"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              version="1.2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18.92 8.776c-.329 1.929-1.211 3.758-2.649 5.48-1.436 1.725-2.71 2.957-3.819 3.695-.699.331-1.293.34-1.786.034-.493-.31-.883-.751-1.169-1.325-.165-.328-.565-1.569-1.202-3.728-.636-2.155-1.017-3.315-1.139-3.479-.083-.163-.288-.184-.616-.061-.33.122-.555.226-.678.309-.123.081-.226.165-.308.245l-.554-.737.616-.74c.452-.492 1.026-1.007 1.724-1.54.7-.534 1.314-.862 1.848-.987.371-.08.679-.028.924.156.247.184.452.484.616.894.165.409.289.811.369 1.199.083.392.165.854.248 1.387.081.534.164.945.246 1.232.451 1.93.821 2.896 1.109 2.896.41 0 1.067-.863 1.971-2.59.41-.779.432-1.426.062-1.941-.369-.512-.943-.522-1.724-.029.123-.78.472-1.456 1.046-2.031 1.026-1.109 2.157-1.521 3.388-1.234 1.273.247 1.765 1.213 1.477 2.895z"
              />
            </svg>
          </a>
        </li>
        <li>
          <a
            href="https://www.pinterest.com/"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              baseProfile="tiny"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              version="1.2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12.486 4.771c-4.23 0-6.363 3.033-6.363 5.562 0 1.533.581 2.894 1.823 3.401.205.084.387.004.446-.221l.182-.717c.061-.221.037-.3-.127-.495-.359-.422-.588-.972-.588-1.747 0-2.25 1.683-4.264 4.384-4.264 2.392 0 3.706 1.463 3.706 3.412 0 2.568-1.137 4.734-2.824 4.734-.932 0-1.629-.77-1.405-1.715.268-1.13.786-2.347.786-3.16 0-.729-.392-1.336-1.2-1.336-.952 0-1.718.984-1.718 2.304 0 .841.286 1.409.286 1.409l-1.146 4.852c-.34 1.44-.051 3.206-.025 3.385.013.104.149.131.21.051.088-.115 1.223-1.517 1.607-2.915.111-.396.627-2.445.627-2.445.311.589 1.213 1.108 2.175 1.108 2.863 0 4.804-2.608 4.804-6.103-.003-2.64-2.24-5.1-5.64-5.1z"
              />
            </svg>
          </a>
        </li>
      </ul>
      <span
        className="text-xs"
      >
        Â© 2022, RENT ME
      </span>
    </div>
  </nav>
</div>
`;
